optuna_dashboard/preferential/_study.py

```diff
@@ -295,21 +273,32 @@ def should_generate(self) -> bool:
         to generate a new trial if this method returns :obj:`True`, and to wait for human
         evaluation if this method returns :obj:`False`.
         """
-        return len(self.best_trials) < get_n_generate(self._study.system_attrs)
+        study_system_attrs = self._study._storage.get_study_system_attrs(
+            self._study._study_id
+        )  # Must come before _study.get_trials()
+        trials = self._study.get_trials(deepcopy=False)
+
+        all_trial_ids = {t._trial_id for t in trials}
+        bad_trial_ids = {
+            trials[worse]._trial_id for (_, worse) in get_preferences(study_system_attrs)
+        }
+        skipped_trial_ids = set(get_skipped_trial_ids(study_system_attrs))
+
+        active_trial_ids = all_trial_ids - bad_trial_ids - skipped_trial_ids
+        return len(active_trial_ids) < get_n_generate(self._study.system_attrs)
```

Comment:
How about rewriting this logic like below?

```suggestion
        trials = self._study.get_trials(deepcopy=False, states=(TrialState.COMPLETE, TrialState.RUNNING))
        worse_trial_numbers = {worse for _, worse in get_preferences(study_system_attrs)}
        skipped_trial_ids = set(get_skipped_trial_ids(study_system_attrs))
        active_trials = [t for t in trials if t.number not in worse_trial_number and t._trial_id not in skipped_trial_ids]
        return len(active_trials) < get_n_generate(self._study.system_attrs)
```

Or maybe we can write this logic like below.

```suggestion
        trials = self._study.get_trials(deepcopy=False, states=(TrialState.COMPLETE, TrialState.RUNNING))
        non_active_trial_numbers = {worse for _, worse in get_preferences(study_system_attrs)}
        non_active_trial_numbers += set(get_skipped_trial_numbers(study_system_attrs))
        active_trials = [t for t in trials if t.number not in non_active_trial_numbers]
        return len(active_trials) < get_n_generate(self._study.system_attrs)
```